test/src/test/java/org/corfudb/runtime/collections/PersistentCorfuTableTest.java

```diff
@@ -422,124 +328,122 @@ private void snapshotRead(long ts, int low, int high) {
+                assertThat(writerResult.get()).isEqualTo(payload2.getLsb());
+            });
         });
-
-        readerThread.start();
-        writerThread.start();
-        readerThread.join();
-        writerThread.join();
-
-        assertThat(readerResult.get()).isEqualTo(payload1.getLsb());
-        assertThat(writerResult.get()).isEqualTo(payload2.getLsb());
     }
 
     /**
      * For MVO instances, the ObjectOpenOption.NO_CACHE should ensure that the instance
      * is not saved in ObjectsView.objectCache or MVOCache.objectCache
      */
     @Test
-    public void testNoCacheOption() {
+    public void testNoCacheOption() throws Exception {
         addSingleServer(SERVERS.PORT_0);
-        rt = getNewRuntime(CorfuRuntime.CorfuRuntimeParameters.builder()
-                .maxCacheEntries(LARGE_CACHE_SIZE)
-                .build())
-                .parseConfigurationString(getDefaultConfigurationString())
-                .connect();
-        setupSerializer();
-
-        UUID streamA = UUID.randomUUID();
-        UUID streamB = UUID.randomUUID();
 
-        PersistentCorfuTable<String, String> tableA = rt.getObjectsView()
-                .build()
-                .setStreamID(streamA)
-                .setTypeToken(new TypeToken<PersistentCorfuTable<String, String>>() {})
-                .addOpenOption(ObjectOpenOption.CACHE)
-                .open();
+        buildNewManagedRuntime(getLargeRtParams(), rt -> {
+            UUID streamA = UUID.randomUUID();
+            UUID streamB = UUID.randomUUID();
 
-        PersistentCorfuTable<String, String> tableB = rt.getObjectsView()
-                .build()
-                .setStreamID(streamB)
-                .setTypeToken(new TypeToken<PersistentCorfuTable<String, String>>() {})
-                .addOpenOption(ObjectOpenOption.NO_CACHE)
-                .open();
-
-        String key = "key";
-        String value = "value";
-
-        tableA.insert(key, value);
-        tableB.insert(key, value);
-
-        // Access the table and populate the cache
-        tableA.size();
-        tableB.size();
-
-        assertThat(rt.getObjectsView().getObjectCache()).containsOnlyKeys(
-                new ObjectsView.ObjectID(streamA, PersistentCorfuTable.class));
-
-        Set<VersionedObjectIdentifier> allKeys = rt.getObjectsView().getMvoCache().keySet();
-        Set<UUID> allObjectIds = allKeys.stream().map(VersionedObjectIdentifier::getObjectId).collect(Collectors.toSet());
-        assertThat(allObjectIds).containsOnly(streamA);
+            try (PersistentCorfuTable<String, String> tableA = rt.getObjectsView()
+                    .build()
+                    .setStreamID(streamA)
+                    .setTypeToken(PersistentCorfuTable.<String, String>getTypeToken())
+                    .addOpenOption(ObjectOpenOption.CACHE)
+                    .open()) {
+
+                try (PersistentCorfuTable<String, String> tableB = rt.getObjectsView()
```

Comment:
Why can't the ManagedCorfuTableForTest class be used here?

Reply:
yes, and I will refactor this code in the next pr